internal/expr/parser.go

```diff
@@ -417,14 +416,23 @@ func (p *Parser) parseGoFullName() (fullName, bool, error) {
 
 // parseList takes a parsing function that returns a fullName and parses a
 // bracketed, comma seperated, list.
-func (p *Parser) parseList(parseFn func(p *Parser) (fullName, bool, error)) ([]fullName, bool, error) {
+func (p *Parser) parseList(singletons bool, parseFn func(p *Parser) (fullName, bool, error)) ([]fullName, bool, error) {
```

Comment:
Bools are typically not a great API for these cases. Just imagine reading some random code with
```
parseList(true, ...)
parseList(false, ...)
```
It's unclear what either mean.

But there's also a second issue here which is the meaning of the parameter itself. From our conversation, it looks like the intention here is to say "Hey, I want to parse a list, but actually don't parse a list." which is not nice. If it's not a list, the code shoud just not call a list parsing function.

There's also a sideways point coming out of this, which is about how we parse one item or many items in certain places. I suggest we enforce consistency for now, so either `bar AS &Foo.bar`, or `(bar) AS (&Foo.bar)`, but not mixed cases. I'd make that true even for `(bar) AS (&Foo.*)` for now at least.

Reply:
Changed to need either brackets on both sides or no brackets at all. When this PR goes out we will have to notify all current users of this breaking change as it may be unexpected.